R/dataframe.R

```diff
@@ -182,24 +182,63 @@ as_data_frame.matrix <- function(x, ...) {
   as_data_frame(as.data.frame(x))
 }
 
-#' Convert row names to an explicit variable.
+#' Conversion between rownames and a column in data frame
+#'
+#' @details add_rownames: Convert row names to an explicit variable.
 #'
 #' @param df Input data frame with rownames.
 #' @param var Name of variable to use
 #' @export
+#' @rdname rownames
+#' @return add_rownames: c("tbl_df", "tbl", "data.frame")
 #' @examples
 #' tbl_df(mtcars)
 #'
-#' add_rownames(mtcars)
+#' (mtcars_tbl <- add_rownames(mtcars) )
 add_rownames <- function(df, var = "rowname") {
   stopifnot(is.data.frame(df))
 
+  if (var %in% colnames(df) ) {
+    stop(paste("There is a column named", var, "already!"))
+  }
+
   rn <- as_data_frame(setNames(list(rownames(df)), var))
   rownames(df) <- NULL
 
   as_data_frame(cbind(rn, df))
+
+}
+
+
+#' @details use_as_rownames: Inverted operation of add_rownames, returning a data.frame
+#'
+#' @rdname rownames
+#' @return use_as_rownames: data.frame
+#' @export
+#' @examples
+#'
+#' head ( use_as_rownames(mtcars_tbl) )
+#'
+use_as_rownames <- function(df, var = "rowname") {
```

Comment:
After a bit of thought, I think "rownames_to_column()" and "column_to_rownames()" are clearer. Let's keep "add_rownames()" with its current functionality for compatibility.

Also, converting row names to columns and vice versa are operations that are somewhat orthogonal to being a tbl_df or not:

```
> mtcars %>% tbl_df %>% rownames
 [1] "Mazda RX4"           "Mazda RX4 Wag"       "Datsun 710"          "Hornet 4 Drive"      "Hornet Sportabout"   "Valiant"            
 [7] "Duster 360"          "Merc 240D"           "Merc 230"            "Merc 280"            "Merc 280C"           "Merc 450SE"         
...
```

I suggest that the new verbs leave the class "as is", so that add_rownames() becomes tbl_df(rownames_to_column(x)) (or rownames_to_column(tbl_df(x)), these two operations should commute).

Would you like to work on that?